### PR TITLE
Fix/agent review pr gate

### PR DESCRIPTION
**Pull request by gannonh**
## Summary by CodeRabbit

* **New Features**
  * PR verification gate: issues in Agent Review now verify an open PR on the matching branch before dispatch; missing PRs are moved back to In Progress with an explanatory tracker comment.
  * Added ability to create automatic issue comments.

* **Chores**
  * Bumped CLI and desktop package versions; updated TypeScript dev dependencies.
  * Adjusted PR composition source to prefer slice issue body plus optional summary artifacts.

* **Documentation**
  * Removed NOTICE/attribution block.

<h3>Greptile Summary</h3>

This PR adds a Symphony-side PR existence gate for "Agent Review" issues and enhances the Kata CLI `pr-lifecycle` extension with new tools (`kata_review_pr`, `kata_fetch_pr_comments`, `kata_resolve_thread`, `kata_reply_to_thread`, `kata_merge_pr`) and improved `runCreatePr` orchestration. The gate prevents the orchestrator from dispatching `Agent Review` issues that lack an open PR, moving them back to `In Progress` with a descriptive comment.

<h3>Confidence Score: 5/5</h3>

Safe to merge — all findings are minor P2 style/cleanup issues with no correctness or data-integrity impact.

The PR gate logic in Rust is well-structured, injectable for testing, and wired into both fresh-dispatch and retry-dispatch paths. The new CLI tools follow existing patterns with proper error handling. The three Rust unit tests and comprehensive TypeScript Vitest suite cover the main paths. All remaining findings are P2.

No files require special attention for merge safety.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/symphony/src/orchestrator.rs | Adds AgentReviewPrStatus, check_agent_review_pr_status_with (injectable for testability), and enforce_agent_review_pr_gate — called at both fresh-dispatch and retry-dispatch paths. Three unit tests cover valid, missing, and closed PR cases. pr_url in Valid variant is unused dead code. |
| apps/symphony/src/main.rs | Adds GithubOrchestratorPort implementation with adapter caching — clean, follows the same pattern as LinearOrchestratorPort. |
| apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts | Adds Linear document loading, branch-name title prefixing, branch auto-push, and body integrity repair. Well-structured with injectable runtime. Minor: empty-head edge case passes --head '' to gh instead of an explicit guard. |
| apps/cli/src/resources/extensions/pr-lifecycle/index.ts | Registers four new tools (kata_review_pr, kata_fetch_pr_comments, kata_resolve_thread, kata_reply_to_thread, kata_merge_pr). Docstring has a minor duplicate. |
| apps/cli/src/resources/extensions/pr-lifecycle/tests/pr-runner.vitest.test.ts | Comprehensive Vitest test suite covering happy path, push-failed, parse-failed, explicit-ID bypass, body-integrity repair, pre-flight failures, Linear config integration, and error edge cases. |
| apps/symphony/prompts/agent-review.md | Adds PR existence preflight and explicit no-comments-yet guard. |
| apps/symphony/prompts/in-progress.md | Adds publish proof steps before state transition to Agent Review. |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Agent as Agent (in-progress.md)
    participant Linear as Linear Tracker
    participant Symphony as Symphony Orchestrator
    participant GH as GitHub (gh CLI)
    participant Worker as Agent (agent-review.md)

    Agent->>GH: git ls-remote / gh pr view (publish proofs)
    Agent->>Linear: Move issue → Agent Review

    Symphony->>Linear: poll: fetch candidate issues
    Linear-->>Symphony: Issue(state=Agent Review)

    Symphony->>Symphony: enforce_agent_review_pr_gate()
    Symphony->>GH: git branch --show-current
    GH-->>Symphony: branch name
    Symphony->>GH: git ls-remote --exit-code --heads origin branch
    GH-->>Symphony: remote exists?

    alt Remote branch missing or no open PR
        Symphony->>Linear: update_issue_state → In Progress
        Symphony->>Linear: create_issue_comment (Symphony note)
        Symphony-->>Symphony: skip dispatch
    else Valid open PR
        Symphony->>Symphony: dispatch worker task
        Worker->>GH: verify PR (belt-and-suspenders)
        Worker->>GH: gh pr view comments / reviews
        Worker->>Worker: address feedback
        Worker->>Linear: Move issue → Human Review
    end
```

<details><summary><h3>Comments Outside Diff (2)</h3></summary>

1. `apps/cli/src/resources/extensions/pr-lifecycle/index.ts`, line 8 ([link](https://github.com/gannonh/kata/blob/76c12b35942352a4ad33f80a92a481162ac06299/apps/cli/src/resources/extensions/pr-lifecycle/index.ts#L8)) 

   <a href="#">P2</a> **Duplicate phrase in doc-comment**

   "three-stage pre-flight (gh installed, gh authenticated, gh authenticated)" repeats "gh authenticated" — the third check was likely meant to be a different step (e.g. "branch has remote").

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/cli/src/resources/extensions/pr-lifecycle/index.ts
   Line: 8

   Comment:
   **Duplicate phrase in doc-comment**

   "three-stage pre-flight (gh installed, gh authenticated, gh authenticated)" repeats "gh authenticated" — the third check was likely meant to be a different step (e.g. "branch has remote").

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

2. `apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts`, line 284-316 ([link](https://github.com/gannonh/kata/blob/76c12b35942352a4ad33f80a92a481162ac06299/apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts#L284-L316)) 

   <a href="#">P2</a> **Empty `head` silently passed to `gh pr create`**

   When `milestoneId`/`sliceId` are provided explicitly (bypassing branch parsing) but `getCurrentBranch()` returns `null` (e.g. detached HEAD), `head` is `""`. The code skips the `ls-remote` check correctly, but then passes `--head ''` to `gh pr create`, which produces a confusing gh-side error instead of a clear pre-flight message. Adding an explicit guard here would surface a better error:

   ```typescript
   if (!head) {
     return {
       ok: false,
       phase: "branch-parse-failed",
       error: "Could not determine current git branch for --head argument",
       hint: "Ensure you are on a named git branch, or pass milestoneId and sliceId explicitly from a valid branch.",
     };
   }
   ```

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts
   Line: 284-316

   Comment:
   **Empty `head` silently passed to `gh pr create`**

   When `milestoneId`/`sliceId` are provided explicitly (bypassing branch parsing) but `getCurrentBranch()` returns `null` (e.g. detached HEAD), `head` is `""`. The code skips the `ls-remote` check correctly, but then passes `--head ''` to `gh pr create`, which produces a confusing gh-side error instead of a clear pre-flight message. Adding an explicit guard here would surface a better error:

   ```typescript
   if (!head) {
     return {
       ok: false,
       phase: "branch-parse-failed",
       error: "Could not determine current git branch for --head argument",
       hint: "Ensure you are on a named git branch, or pass milestoneId and sliceId explicitly from a valid branch.",
     };
   }
   ```

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/pr-lifecycle/index.ts
Line: 8

Comment:
**Duplicate phrase in doc-comment**

"three-stage pre-flight (gh installed, gh authenticated, gh authenticated)" repeats "gh authenticated" — the third check was likely meant to be a different step (e.g. "branch has remote").

```suggestion
 * The tool runs a two-stage pre-flight (gh installed, gh authenticated),
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts
Line: 284-316

Comment:
**Empty `head` silently passed to `gh pr create`**

When `milestoneId`/`sliceId` are provided explicitly (bypassing branch parsing) but `getCurrentBranch()` returns `null` (e.g. detached HEAD), `head` is `""`. The code skips the `ls-remote` check correctly, but then passes `--head ''` to `gh pr create`, which produces a confusing gh-side error instead of a clear pre-flight message. Adding an explicit guard here would surface a better error:

```typescript
if (!head) {
  return {
    ok: false,
    phase: "branch-parse-failed",
    error: "Could not determine current git branch for --head argument",
    hint: "Ensure you are on a named git branch, or pass milestoneId and sliceId explicitly from a valid branch.",
  };
}
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/symphony/src/orchestrator.rs
Line: 52-61

Comment:
**`pr_url` in `Valid` variant is never read**

`AgentReviewPrStatus::Valid { branch, pr_url }` captures the PR URL, but `enforce_agent_review_pr_gate` only pattern-matches on `Missing` and discards `Valid` entirely — `pr_url` is never consumed. Consider either removing the field or adding a brief comment marking it as reserved for future use (e.g. to surface the URL in dispatch telemetry).

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/symphony/src/orchestrator.rs
Line: 5936-5971

Comment:
**Missing test coverage for several gate branches**

The three existing tests cover "valid", "no PR found", and "closed PR" — but `check_agent_review_pr_status_with` also handles:

- Remote branch missing on `origin` (the `ls-remote` non-success path, line ~149)
- `head_ref_name` mismatch (PR for a different branch, line ~212)
- `base_ref_name` mismatch (PR targeting wrong base, line ~226)

Adding tests for these cases would prevent regressions, especially since the function is used as a hard gate that can demote issues.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["fix(cli): use slice issue description fo..."](https://github.com/gannonh/kata/commit/76c12b35942352a4ad33f80a92a481162ac06299) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28241767)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/symphony/src/orchestrator.rs [hunk 1]
```diff
@@ -39,6 +41,220 @@ static RATE_LIMIT_WINDOW_RE: LazyLock<Regex> = LazyLock::new(|| {
...
+    if !remote_result.success {
+        let detail = if remote_result.stderr.is_empty() {
+            remote_result.stdout
+        } else {
+            remote_result.stderr
+        };
+        return AgentReviewPrStatus::Missing {
+            branch: Some(branch),
+            reason: format!("remote branch is missing on origin ({detail})"),
+        };
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs:208]: Fixed. When both stdout and stderr are empty, the detail now falls back to `"ref not found"` instead of an empty string.

File: apps/symphony/src/orchestrator.rs [hunk 2]
```diff
@@ -4200,6 +4425,60 @@ impl Orchestrator {
...
+                error = %err,
+                "failed to write agent review reset note"
+            );
+        }
+
+        let refreshed = port.refresh_issue(&issue.id)?.unwrap_or_else(|| {
+            let mut demoted = issue.clone();
+            demoted.state = "In Progress".to_string();
+            demoted
+        });
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs]: Fixed. `enforce_agent_review_pr_gate` now returns `Ok(None)` after demotion, preventing dispatch in the same cycle. The `refresh_issue` call and stale-state concern are eliminated entirely.

File: apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts [hunk 3]
```diff
@@ -244,13 +242,13 @@ export async function runCreatePr(options: PrCreateOptions): Promise<PrCreateRes
...
+  // Ensure we have the required slice plan from the slice issue description.
   if (!linearDocuments?.PLAN) {
     return {
       ok: false,
       phase: "artifact-error",
-      error: `Missing required Linear artifact: ${sliceId}-PLAN`,
-      hint: `Create or fetch ${sliceId}-PLAN before creating a PR.`,
+      error: `Missing required slice issue description for ${sliceId}`,
+      hint: `Populate the ${sliceId} slice issue description before creating a PR.`,
     };
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts]: Fixed. The error message now differentiates three cases: missing Linear config, failed issue resolution, and missing PLAN document. Each case gets a specific error and hint.

File: apps/symphony/src/orchestrator.rs [hunk 4]
```diff
@@ -39,6 +41,220 @@ static RATE_LIMIT_WINDOW_RE: LazyLock<Regex> = LazyLock::new(|| {
...
+enum AgentReviewPrStatus {
+    Valid {
+        branch: String,
+        pr_url: String,
+    },
+    Missing {
+        branch: Option<String>,
+        reason: String,
+    },
+}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs:66]: Keeping `pr_url` in the `Valid` variant for now. It is used in test assertions and will be consumed for dispatch telemetry in a follow-up.

File: apps/symphony/src/orchestrator.rs [hunk 5]
```diff
@@ -5557,4 +5856,117 @@ mod tests {
...
+        );
+
+        assert_eq!(
+            status,
+            AgentReviewPrStatus::Missing {
+                branch: Some("sym/KAT-2499".to_string()),
+                reason: "PR exists but is not open (state: CLOSED)".to_string(),
+            }
+        );
+    }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs:6025]: Fixed. Added tests for all three missing branches: remote branch missing on origin, head_ref_name mismatch, and base_ref_name mismatch. Also added a test for the new `CheckFailed` variant (command timeout/launch failure).

File: apps/symphony/src/orchestrator.rs [hunk 6]
```diff
@@ -39,6 +41,220 @@ static RATE_LIMIT_WINDOW_RE: LazyLock<Regex> = LazyLock::new(|| {
...
+        "gh",
+        &["pr", "view", "--json", "url,state,headRefName,baseRefName"],
+        workspace_path,
+    ) {
+        Ok(result) => result,
+        Err(err) => {
+            return AgentReviewPrStatus::Missing {
+                branch: Some(branch),
+                reason: format!("could not verify open PR: {err}"),
+            };
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs:220]: Fixed. Added a new `AgentReviewPrStatus::CheckFailed` variant. All `Err` paths from `run()` (process launch failure, timeout) now return `CheckFailed` instead of `Missing`. The gate skips the issue without demotion on transient failures.

File: apps/symphony/src/orchestrator.rs [hunk 7]
```diff
@@ -39,6 +41,220 @@ static RATE_LIMIT_WINDOW_RE: LazyLock<Regex> = LazyLock::new(|| {
...
+        .current_dir(cwd)
+        .output()
+        .map_err(|err| format!("{} {} failed to start: {}", program, args.join(" "), err))?;
+
+    Ok(CommandOutcome {
+        success: output.status.success(),
+        stdout: String::from_utf8_lossy(&output.stdout).trim().to_string(),
+        stderr: String::from_utf8_lossy(&output.stderr).trim().to_string(),
+    })
+}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs:139]: Fixed. `run_command_outcome` now sets `GIT_TERMINAL_PROMPT=0` and `GH_PROMPT_DISABLED=1` on all subprocess invocations, and implements a 30-second timeout via `try_wait` polling. Timed-out processes are killed and return `Err`, which maps to the new `CheckFailed` variant (no demotion).

File: apps/symphony/src/orchestrator.rs [hunk 8]
```diff
@@ -4200,6 +4425,60 @@ impl Orchestrator {
...
+            );
+        }
+
+        let refreshed = port.refresh_issue(&issue.id)?.unwrap_or_else(|| {
+            let mut demoted = issue.clone();
+            demoted.state = "In Progress".to_string();
+            demoted
+        });
+
+        Ok(Some(refreshed))
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/symphony/src/orchestrator.rs]: Fixed. `enforce_agent_review_pr_gate` now returns `Ok(None)` after demotion instead of `Ok(Some(refreshed))`. The `refresh_issue` call is removed entirely, so dispatch cannot proceed for a demoted issue.

File: apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts]: Fixed. The error messaging now differentiates based on whether Linear config is present, whether the issue was resolved, or whether the PLAN document is missing. Each case provides a workflow-appropriate hint.
